src/formatDate.js

```diff
@@ -50,7 +50,59 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const switchYearFormat = () => {
+    if (year.length === 2) {
+      if (+year < 30) {
+        year = '20' + year;
+      } else {
+        year = '19' + year;
+      }
+    } else {
+      year = year.slice(2);
+    }
+  };
+
+  let year, month, day, yearIndex;
+  const dateItems = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        year = dateItems[i];
+        yearIndex = i;
+        break;
+      case 'MM':
+        month = dateItems[i];
+        break;
+      case 'DD':
+        day = dateItems[i];
+        break;
+    }
+  }
+
+  if (!toFormat.includes(fromFormat[yearIndex])) {
+    switchYearFormat();
+  }
+
+  const newDate = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        newDate.push(year);
+        break;
+      case 'MM':
+        newDate.push(month);
+        break;
+      case 'DD':
+        newDate.push(day);
+        break;
+    }
```

Comment:
Can be replaced with just `newDate.push(dateObject[toFormat[i]]);` if you make some optimizations.